### DISCUSSION TITLE
Example of dynamically constructed tiles.

### DISCUSSION THREAD

**Discussion by rob-myers** [isaac-mason/recast-navigation-js#316] [General]
Do you have an example where Tiles are created dynamically i.e. some constructed initially, some constructed later?

If not I will try to create one, although some guidance would be appreciated.

**Comment by isaac-mason** [comment]
I do on sketches.isaacmason.com, but I need to create a minimal example here that doesn't have a bunch of other stuff going on 😄 

See:
- https://sketches.isaacmason.com/sketch/recast-navigation/dynamic-tiled-navmesh
- https://github.com/isaac-mason/sketches/tree/main/src/sketches/recast-navigation/dynamic-tiled-navmesh

**Reply by isaac-mason** [reply]
This also generates navmesh tiles in a web worker, and regenerates tiles with active physics bodies in them.

Ideally we'd have a example/storybook demonstrating just the fundamentals without all the other stuff - rapier physics, react three fiber, web workers, etc.

**Reply by rob-myers** [reply]
Thanks for the example. So [whenever an entity is added](https://github.com/isaac-mason/sketches/blob/8b449d77495441248a9fd32041ce2e0cdc585815/src/sketches/recast-navigation/dynamic-tiled-navmesh/navigation/navigation.tsx#L91), you:
- check which tiles it intersects with,
- rebuild those tiles (in a web worker),
- replaces tiles [here](https://github.com/isaac-mason/sketches/blob/8b449d77495441248a9fd32041ce2e0cdc585815/src/sketches/recast-navigation/dynamic-tiled-navmesh/navigation/dynamic-tiled-navmesh.ts#L94).

If so, this is exactly what I was looking for.
I'm happy to extract an example. Good way to understand what you're doing.

**Reply by isaac-mason** [reply]
Yep that's the core of it. I also rebuild tiles with [active rigid bodies](https://github.com/isaac-mason/sketches/blob/8b449d77495441248a9fd32041ce2e0cdc585815/src/sketches/recast-navigation/dynamic-tiled-navmesh/navigation/navigation.tsx#L123).

It's worth calling out that the approach can be improved too. In that demo, for any tile rebuild I send the web worker positions and indices for _everything in the scene_, then in the web worker narrow down to triangles for the tile bounds. Because in that demo we just have one simple level mesh, this works ok.

If we wanted a dynamic navmesh for a large level, ideally the level geometry would be in parts/tiles, and we'd only send the web worker positions and indices for meshes within the bounds of the tile we're rebuilding.
